app/react/Library/components/TableRow.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { PropertySchema } from 'shared/types/commonTypes';
+import { TemplateSchema } from 'shared/types/templateType';
+import { ThesaurusSchema } from 'shared/types/thesaurusType';
+import formatter from 'app/Metadata/helpers/formater';
+import MarkdownViewer from 'app/Markdown';
+import { I18NLink } from 'app/I18N';
+import GeolocationViewer from 'app/Metadata/components/GeolocationViewer';
+
+interface TableRowProps {
+  columns: PropertySchema[];
+  document: any;
+  storeKey: 'library' | 'uploads';
+  selected?: boolean;
+  onClick: any;
+  templates: any;
+  thesauris: any;
+  zoomLevel: number;
+}
+
+function getLink(url: string, label: string) {
+  return (
+    <I18NLink key={url} to={url}>
+      {label}
+    </I18NLink>
+  );
+}
+
+function formatProperty(prop: any) {
+  let result = prop?.value;
+  if (!result) {
+    return typeof prop === 'string' ? prop : undefined;
+  }
+  if (['date', 'daterange', 'numeric', 'select', 'text', undefined].includes(prop.type)) {
+    return result;
+  }
+
+  switch (prop.type) {
+    case 'multiselect':
+    case 'multidaterange':
+    case 'multidate':
+      result = prop.value.map((p: any) => p.value).join(', ');
+      break;
+    case 'markdown':
+      result = <MarkdownViewer markdown={prop.value} />;
+      break;
+    case 'link':
+      result = getLink(prop.value.url, prop.value.label);
+      break;
+    case 'relationship':
+      result = prop.value.map((p: any, index: number) => (
+        <React.Fragment key={index}>
+          {index > 0 && ', '}
+          {getLink(p.url, p.value)}
+        </React.Fragment>
+      ));
+      break;
+    case 'geolocation':
+      result = <GeolocationViewer points={prop.value} onlyForCards />;
+      break;
+    default:
+      result = undefined;
+      break;
+  }
+  return result;
+}
+
+function formatDocument(document: any, templates: TemplateSchema[], thesauris: ThesaurusSchema[]) {
+  let formattedDoc = document.toJS();
+  const template = templates.find((t: TemplateSchema) => t.get('_id') === formattedDoc.template);
+  formattedDoc.templateName = template?.get('name');
+  formattedDoc = formatter.prepareMetadata(formattedDoc, templates, thesauris, null, {
+    sortedProperty: 'creationDate',
+  });
+  formattedDoc.metadata.forEach((prop: any) => {
+    formattedDoc[prop.name] = prop;
+  });
+  return formattedDoc;
+}
+
+function displayCell(
```

Comment:
This should really be a different component name simply Cell.

Don't be afraid to have many smaller files.  It makes for much easier reusability, testing, maintenability.